packages/kbn-test/src/functional_tests/tasks.js

```diff
@@ -24,116 +20,89 @@ in another terminal session by running this command from this directory:
+    log.error(`startServers requires configPath`);
+    process.exit(1);
   }
 
-  const configOption = getopts(process.argv.slice(2)).config;
-
-  configPath = await resolve(KIBANA_ROOT, configOption || configPath);
+  configPath = resolve(process.cwd(), configPath);
 
   try {
-    await withTmpDir(async tmpDir => {
-      await withProcRunner(async procs => {
-        const config = await readConfigFile(log, configPath);
-        const { runEs, runKbn } = options;
-
-        const es = await runEs({ tmpDir, config });
-        await runKbn({ procs, config });
-
-        // wait for 5 seconds of silence before logging the success message
-        // so that it doesn't get buried
-        await Rx.Observable.fromEvent(log, 'data')
-          .switchMap(() => Rx.Observable.timer(5000))
-          .first()
-          .toPromise();
-
-        log.info(SUCCESS_MESSAGE);
-        await procs.waitForAllToStop();
-        await es.cleanup();
-      });
+    await withProcRunner(log, async procs => {
+      const config = await readConfigFile(log, configPath);
+
+      const es = await runElasticsearch({ config, log });
+      await runKibanaServer({ procs, config, log });
+
+      // wait for 5 seconds of silence before logging the success message
+      // so that it doesn't get buried
+      await Rx.Observable.fromEvent(log, 'data')
+        .switchMap(() => Rx.Observable.timer(5000))
+        .first()
+        .toPromise();
+
+      log.info(SUCCESS_MESSAGE);
+      await procs.waitForAllToStop();
+      await es.cleanup();
     });
   } catch (err) {
     fatalErrorHandler(err);
   }
 }
 
-
 /*
  * Start servers and run tests for single config
  */
-async function runSingleConfig(
-  configPath,
-  options = {
-    runEs: runElasticsearch,
-    runKbn: runKibanaServer,
-  }
-) {
-  try {
-    await withTmpDir(async tmpDir => {
-      await withProcRunner(async procs => {
-        const config = await readConfigFile(log, configPath);
-        const { runEs, runKbn } = options;
-
-        const es = await runEs({ tmpDir, config });
-        await runKbn({ procs, config });
-        await runFtr({ procs, configPath, cwd: process.cwd() });
-
-        await procs.stop('kibana');
-        await es.cleanup();
-      });
+async function runSingleConfig(configPath, { bail, log }) {
+  await withProcRunner(log, async procs => {
+    const config = await readConfigFile(log, configPath);
+
+    const es = await runElasticsearch({ config, log });
+    await runKibanaServer({ procs, config });
+
+    // Note: When solving how to incorporate functional_test_runner
+    // clean this up
+    const logLevel = log.getLevel();
+    await runFtr({
+      procs,
+      configPath,
+      bail,
+      logLevel,
+      cwd: process.cwd(),
     });
-  } catch (err) {
-    fatalErrorHandler(err);
-  }
-}
 
+    await procs.stop('kibana');
+    await es.cleanup();
+  });
+}
 
-function fatalErrorHandler(err) {
+function fatalErrorHandler(err, { log }) {
```

Comment:
I think things like exiting the process and setting the exit code is the responsibility of the CLI functions, not the internal functions. They should produce rejected promises.